Update VERSIONS_YAML_FILE e2e test information (#916)
istio-ecosystem/sail-operator: README.md

#### versions.yaml

> The name of the versions.yaml file can be overwritten using the VERSIONS_YAML_FILE environment variable. This way, downstream vendors can point to a custom list of supported versions. take into account that this file should be located in the `pkg/istioversion` directory, so the default value is `pkg/istioversion/versions.yaml`.

Reviewer: ```suggestion
The name of the versions.yaml file can be overwritten using the VERSIONS_YAML_FILE environment variable. This way, downstream vendors can point to a custom list of supported versions. Note that this file should be located in the `pkg/istioversion` directory, with the default value being `pkg/istioversion/versions.yaml`.
```

> The name of the versions.yaml file can be overwritten using the VERSIONS_YAML_FILE environment variable. This way, downstream vendors can point to a custom list of supported versions. take into account that this file should be located in the `pkg/istioversion` directory, so the default value is `pkg/istioversion/versions.yaml`.

Author: Hey @FilipB, but `VERSIONS_YAML_DIR` is just used in the chart generation and removal of old istio charts, but if you try to run the test with this var, we will get this error, for example:
```
2025-06-18T10:36:27.306192Z	info	loading supported istio versions from customversion.yaml
panic: failed to read versions from 'customversion.yaml': open customversion.yaml: file does not exist
```

This is because of the version.go package is using embed to get the version yaml file and is using only the env var `VERSIONS_YAML_FILE` when it is init
```
func init() {
	// we can't use ldflags when running tests, use an env variable instead
	// tmp workaround for https://github.com/golang/go/issues/64246
	if versionsFilename == "" {
		versionsFilename = env.Get("VERSIONS_YAML_FILE", "versions.yaml")
	}
	log.Info("loading supported istio versions from " + versionsFilename)
	// Read the embedded versions.yaml file
	data, err := versionsFiles.ReadFile(versionsFilename)
	if err != nil {
		panic(fmt.Errorf("failed to read versions from '%s': %w", versionsFilename, err))
	}

	List, Default, Base, New, Map, aliasList, EOL = mustParseVersionsYaml(data)
}
```

Go disallows other paths in //go: embed paths, to be able to use it as we need to place it in the same directory as the go function

> The name of the versions.yaml file can be overwritten using the VERSIONS_YAML_FILE environment variable. This way, downstream vendors can point to a custom list of supported versions. take into account that this file should be located in the `pkg/istioversion` directory, so the default value is `pkg/istioversion/versions.yaml`.

Author: I'll rephrase the sentence in the doc to clarify that the file must be located there and we just need to provide the file name
